### PR TITLE
[FLINK-34000] Implement restore tests for IncrementalGroupAgg node

### PR DESCRIPTION
**Pull request by bvarghese1**
## What is the purpose of the change

*Add restore tests for IncrementalGroupAggregate node*

## Verifying this change
This change added tests and can be verified as follows:

- Added restore tests for IncrementalGroupAggregate node which verifies the generated compiled plan with the saved compiled plan.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/IncrementalGroupAggregateTestPrograms.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema(SOURCE_SCHEMA)
+                                    .producedBeforeRestore(BEFORE_DATA)
+                                    .producedAfterRestore(AFTER_DATA)
+                                    .build())
+                    .setupConfig(ExecutionConfigOptions.TABLE_EXEC_MINIBATCH_ENABLED, true)
+                    .setupConfig(
+                            ExecutionConfigOptions.TABLE_EXEC_MINIBATCH_ALLOW_LATENCY,
+                            Duration.ofSeconds(10))
+                    .setupConfig(ExecutionConfigOptions.TABLE_EXEC_MINIBATCH_SIZE, 1L)
```
Comment (bvarghese1) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/IncrementalGroupAggregateTestPrograms.java:89]: @dawidwys I set the `TABLE_EXEC_MINIBATCH_SIZE` to `1` to make the results more predictable.

**Review by dawidwys — APPROVED** [review]
LGTM, but we should check if the e2e test failures are unrelated

**Comment by dawidwys** [comment]
@bvarghese1 Do you mind rebasing on top of the current master? I believe failures should be fixed now: https://issues.apache.org/jira/browse/FLINK-34420

**Comment by bvarghese1** [comment]
@dawidwys Rebased.

**Pull request merged**